source/core/Variants.cpp

```diff
@@ -0,0 +1,191 @@
+
+            auto pairIterBool = attributeMap->insert(VariantAttributeManager::AttributeMapType::value_type(name, value));
+            _Param(Replaced) = !pairIterBool.second;
+            if (!pairIterBool.second) {
+                pairIterBool.first->second = value;
+            }
+        } else {
+            VariantAttributeManager::AttributeMapType *attributeMap = new VariantAttributeManager::AttributeMapType;
+            (*attributeMap)[name] = value;
+            variantToAttributeMap[inputVariant] = attributeMap;
+            _Param(Replaced) = false;
+        }
+    }
+    return _NextInstruction();
+}
+
+struct GetVariantAttributesParamBlock : public VarArgInstruction
+{
+    _ParamDef(TypeRef, InputVariant);
+    _ParamDef(StringRef, Name);
+    _ParamImmediateDef(StaticTypeAndData, Value);
+    _ParamDef(Boolean, Found);
+    _ParamDef(ErrorCluster, ErrorClust);
+    NEXT_INSTRUCTION_METHODV()
+};
+
+VIREO_FUNCTION_SIGNATUREV(GetVariantAttributes, GetVariantAttributesParamBlock)
+{
+    ErrorCluster *errPtr = _ParamPointer(ErrorClust);
+    _Param(Found) = false;
+    if (!errPtr || !errPtr->status) {
+        TypeRef inputVariant = _Param(InputVariant);
+        StringRef name = _Param(Name);
+        StaticTypeAndDataPointer value = &_ParamImmediate(Value);
+        const VariantAttributeManager::VariantToAttributeMapType &variantToAttributeMap = VariantAttributeManager::Instance().GetVariantToAttributeMap();
+
+        auto variantToAttributeMapIter = variantToAttributeMap.find(inputVariant);
+        if (variantToAttributeMapIter != variantToAttributeMap.end()) {
+            VariantAttributeManager::AttributeMapType *attributeMap = variantToAttributeMapIter->second;
+            auto attributeMapIter = attributeMap->find(name);
+            if (attributeMapIter != attributeMap->end()) {
+                StaticTypeAndDataPointer foundValue = attributeMapIter->second;
+                if (foundValue->_paramType->IsA(value->_paramType)) {
+                    _Param(Found) = true;
+                    value->_paramType->CopyData(foundValue->_pData, value->_pData);
+                } else {
+                    errPtr->SetError(true, 91, "Get Variant Attribute");  // Incorrect type for default value for the attribute
+                }
+            }
+        }
+    }
+    return _NextInstruction();
+}
+
+DEFINE_VIREO_BEGIN(Variant)
+
+    DEFINE_VIREO_FUNCTION(VariantToData, "p(i(VarArgCount argumentCount) i(Variant) io(ErrorCluster) o(StaticTypeAndData))");
+    DEFINE_VIREO_FUNCTION(DataToVariant, "p(i(VarArgCount argumentCount) i(StaticTypeAndData) o(Variant))");
+    DEFINE_VIREO_FUNCTION(SetVariantAttributes, "p(i(VarArgCount argCount) io(Variant inputVariant) i(String name)"
+                                                " i(StaticTypeAndData value) o(Boolean replaced) io(ErrorCluster error) )");
+    DEFINE_VIREO_FUNCTION(GetVariantAttributes, "p(i(VarArgCount argCount) io(Variant inputVariant) i(String name)"
```

Comment:
Actually, GetVariantAttribute, although the name is singular for the LabVIEW prim, can return either a single attribute or an array  of them depending on if name is passed.
It would probably be best in Vireo to have two entry points, one singular with passes name and one plural which does not an always returns an array, and have DFIR generate the right one depending on if name is wired.

Reply:
Done for both of the comments in new PR. Thanks.